ksql-common/src/main/java/io/confluent/ksql/function/KsqlFunction.java

```diff
@@ -95,46 +132,13 @@ static KsqlFunction create(
         isVariadic);
   }
 
-  private KsqlFunction(
-      final Schema returnType,
-      final List<Schema> arguments,
-      final String functionName,
-      final Class<? extends Kudf> kudfClass,
-      final Function<KsqlConfig, Kudf> udfFactory,
-      final String description,
-      final String pathLoadedFrom,
-      final boolean isVariadic) {
-    this.returnType = Objects.requireNonNull(returnType, "returnType");
-    this.parameters = ImmutableList.copyOf(Objects.requireNonNull(arguments, "arguments"));
-    this.functionName = Objects.requireNonNull(functionName, "functionName");
-    this.kudfClass = Objects.requireNonNull(kudfClass, "kudfClass");
-    this.udfFactory = Objects.requireNonNull(udfFactory, "udfFactory");
-    this.description = Objects.requireNonNull(description, "description");
-    this.pathLoadedFrom  = Objects.requireNonNull(pathLoadedFrom, "pathLoadedFrom");
-    this.isVariadic = isVariadic;
-    this.hasGenerics = GenericsUtil.hasGenerics(returnType);
-
-    if (arguments.stream().anyMatch(Objects::isNull)) {
-      throw new IllegalArgumentException("KSQL Function can't have null argument types");
-    }
-    if (isVariadic) {
-      if (arguments.isEmpty()) {
-        throw new IllegalArgumentException(
-            "KSQL variadic functions must have at least one parameter");
-      }
-      if (!Iterables.getLast(arguments).type().equals(Type.ARRAY)) {
-        throw new IllegalArgumentException(
-            "KSQL variadic functions must have ARRAY type as their last parameter");
-      }
-    }
+  public Schema getReturnType(final List<Schema> arguments) {
 
-    if (!returnType.isOptional()) {
-      throw new IllegalArgumentException("KSQL only supports optional field types");
+    if (returnSchemaProvider == null) {
```

Comment:
Did you intended to put this check below line 141? where it currently is, this check will never be true because in the constructor we ensure that the `returnSchemaProvider` is never null.

If you intended to check the return type, then please add a test for it!